src/plugins/dashboard/public/dashboard_app/dashboard_app.tsx

```diff
@@ -188,6 +190,24 @@ export function DashboardApp({
     getScreenshotContext,
   ]);
 
+  useMemo(() => {
+    if (!dashboardAPI) return;
+    dashboardAPI?.expandedPanelId.subscribe(() => {
```

Comment:
This subscription seems to be firing **on load** of a dashboard with... the dashboard ID for some reason? Even when there is no expanded panel ID. To see this, print out the `expandedPanelId` in this subscription and refresh the dashboard.

I think we should fix this before this is ready to merge, because this is causing the URL to "blip" on load 🙇

Reply:
I might need some help understanding this, but with the dashboard_router.tsx on line 161 we are getting the expandedPanelId from the URL or falling into line 162 where we would be getting a url without the /:expandedPanelId but with the /:id of the dashboard.

Reply:
Yup, and that part seems to be working!! :) I am specifically talking about the fact that the `dashboardAPI?.expandedPanelId` is emitting the `dashboardId` on first load - it has nothing to do with the router. The reason you are seeing the router **respond** to this is because of the `history.replace` stuff - if you comment out that logic and simply `console.log(expandedPanelId)`, you'll see the unnecessary emit. 

LMK if that's not clear!!

Reply:
Testing locally, I see undefined being logged (not the dashboard id 👍🏻) and then the expandedPanelId when a panel is maximized. I think this is working after [c8cfcdd](https://github.com/elastic/kibana/pull/190086/commits/c8cfcdd10bd7c3b0730b1ffc61a2215532420b00)!!!

Reply:
@rshen91 I'm still seeing it 🙇 


https://github.com/user-attachments/assets/726be5f9-6e81-4061-bd64-8468049666ed

And this is causing the URL to blip, still:


https://github.com/user-attachments/assets/6f207415-600a-4cf4-a631-e7b7134f4579

Reply:
You rock thank you for zooming with me and going through this. For my own learning, there was a part in initializing the expandedPanelId in dashboard_container where we set the behavior subject for the expandedPanelId to the SavedObject id/Dashboard Id. This should not have been the case so setting that initial value to undefined resolved it!